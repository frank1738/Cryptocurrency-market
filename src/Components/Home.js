import React from 'react';
import { useDispatch } from 'react-redux';
import { useEffect } from 'react';
import { fetchCurrencies } from '../Redux/Currencies/currencies';
import { useSelector } from 'react-redux';
import { NavLink } from 'react-router-dom';
import Coin from './Coin';
import Search from './Search';

const Home = ({ page, set }) => {
  const data = useSelector((state) => state.coins);
  const dispatch = useDispatch();
  useEffect(() => {
    if (data.length === 0) dispatch(fetchCurrencies());
  }, [dispatch]);

  return (
    <div className="container">
      <Search />
      <div>
        <div className="heading">
          <p className="first title">#</p>
          <p className="width title">Coin</p>
          <p className="width title">Price</p>
          <p className="width title">24h</p>
          <p className="hide-mobile width title">Volume</p>
          <p className="hide-mobile width title">Mkt Cap</p>
        </div>
        {data.length > 0 ? (
          data.map((item) => (
            <NavLink
              to={`/coin/${item.id}`}
              className="link"
              key={item.id}
              onClick={() => set('coins')}
            >
              <Coin
                id={item.id}
                marketRank={item.market_cap_rank}
                image={item.image}
                coinSymbol={item.symbol}
                currentPrice={item.current_price}
                priceChange={item.price_change_percentage_24h}
                totalVolume={item.total_volume}
                marketCap={item.market_cap}
                imageUrl={item.image}
                coinName={item.name}
              />
            </NavLink>
          ))
        ) : (
          <div className="warning">
            <i className="uil uil-exclamation-triangle warn-icon"></i>
            <h1>Coin Not Available</h1>{' '}
          </div>
        )}
      </div>
    </div>
  );
};

export default Home;
