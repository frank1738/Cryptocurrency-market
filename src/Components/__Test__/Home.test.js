import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import Home from '../Home';
import { store } from '../../Redux/configureStore';

describe('Home page test', () => {
  test('should render home page properly', () => {
    const component = render(
      <Provider store={store}>
        <Home />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
