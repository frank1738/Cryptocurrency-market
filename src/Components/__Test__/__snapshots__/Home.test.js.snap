// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page test should render home page properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="search-container"
        >
          <input
            class="input"
            placeholder="Search"
            type="text"
          />
        </div>
        <div>
          <div
            class="heading"
          >
            <p
              class="first title"
            >
              #
            </p>
            <p
              class="width title"
            >
              Coin
            </p>
            <p
              class="width title"
            >
              Price
            </p>
            <p
              class="width title"
            >
              24h
            </p>
            <p
              class="hide-mobile width title"
            >
              Volume
            </p>
            <p
              class="hide-mobile width title"
            >
              Mkt Cap
            </p>
          </div>
          <div
            class="warning"
          >
            <i
              class="uil uil-exclamation-triangle warn-icon"
            />
            <h1>
              Coin Not Available
            </h1>
             
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="search-container"
      >
        <input
          class="input"
          placeholder="Search"
          type="text"
        />
      </div>
      <div>
        <div
          class="heading"
        >
          <p
            class="first title"
          >
            #
          </p>
          <p
            class="width title"
          >
            Coin
          </p>
          <p
            class="width title"
          >
            Price
          </p>
          <p
            class="width title"
          >
            24h
          </p>
          <p
            class="hide-mobile width title"
          >
            Volume
          </p>
          <p
            class="hide-mobile width title"
          >
            Mkt Cap
          </p>
        </div>
        <div
          class="warning"
        >
          <i
            class="uil uil-exclamation-triangle warn-icon"
          />
          <h1>
            Coin Not Available
          </h1>
           
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
